src/main/java/tasks/TaskManager.java

```diff
@@ -27,36 +32,28 @@
 	public TaskManager(FBase fBase) {
 		this.fBase = fBase;
 		pool = Executors.newCachedThreadPool();
-		runningTasks = new HashMap<TaskName, Integer>();
+		for (int i = 0; i < runningTasks.length; i++)
+			runningTasks[i] = new AtomicInteger(0);
 	}
-	
-	public synchronized void registerTask(TaskName name) {
-		Integer value = runningTasks.get(name);
-		if (value != null) {
-			runningTasks.put(name, value + 1);
-		} else {
-			runningTasks.put(name, 1);
-		}
+
+	public void registerTask(TaskName name) {
+		runningTasks[name.ordinal()].incrementAndGet();
 	}
 
-	public synchronized void deregisterTask(TaskName name) {
+	public void deregisterTask(TaskName name) {
 		logger.debug("Deregistering task " + name);
-		if (runningTasks.containsKey(name)) {
-			int number = runningTasks.get(name) - 1;
-			if (number > 0) {
-				runningTasks.put(name, number);
-			} else {
-				runningTasks.remove(name);
-			}
-		}
+		runningTasks[name.ordinal()].decrementAndGet();
 	}
 
-	public synchronized HashMap<TaskName, Integer> getRunningTaskNumbers() {
-		return new HashMap<TaskName, Integer>(runningTasks);
+	
+	public Map<TaskName, Integer> getRunningTaskNumbers() {
```

Comment:
I think it might be useful for admins to check which tasks are not finishing or see the current state. However, this is not going to happen to often so we might be fine 👍